fdbclient/VersionVector.h

```diff
@@ -83,33 +86,38 @@ struct VersionVector {
 			return; // rerurn an invalid version vector
 		}
 
-		std::map<Version, std::set<Tag>> versionMap; // order versions
+		std::map<Version, std::set<Tag>> tmpVersionMap; // order versions
 		for (const auto& [tag, version] : versions) {
 			if (version > refVersion) {
-				versionMap[version].insert(tag);
+				tmpVersionMap[version].insert(tag);
 			}
 		}
 
-		for (auto& [version, tags] : versionMap) {
+		for (auto& [version, tags] : tmpVersionMap) {
 			delta.setVersion(tags, version);
 		}
 	}
 
+	// @note this method, together with method getDelta(), helps minimize
+	// the number of version vector entries that get sent from sequencer to
+	// grv proxy (and from grv proxy to client) on the read path.
 	void applyDelta(const VersionVector& delta) {
 		if (delta.maxVersion == invalidVersion) {
 			return;
 		}
 
-		ASSERT(maxVersion < delta.maxVersion);
```

Comment:
Simulation tests were triggering this assert. The assertion failure was because a client doesn't need to make GetReadVersion requests sequentially, so there could be an overlap in the deltas received for successive read version requests.